layers.go

```diff
@@ -456,6 +463,7 @@ func copyLayer(l *Layer) *Layer {
 		CompressionType:    l.CompressionType,
 		ReadOnly:           l.ReadOnly,
 		volatileStore:      l.volatileStore,
+		imageStore:         l.imageStore,
```

Comment:
I think using the location index and/or enum directly would ultimately be simpler than 2 separate bools. Now we have to worry about both being set.

Reply:
I'll change to use directly the location in the layer, but we will still need to handle the two bools that we get from the options (since Volatile is part of the API at this point)

Reply:
Well… we _could_ push that decision all the way to `store.go`, but really https://github.com/containers/storage/pull/2258#discussion_r1958485813 needs to be decided first.

As is, I think one location mapping from bools to the enum, and the simplification in other places, is worth it already.